app/src/main/java/com/github/geohunt/app/model/database/Database.kt

```diff
@@ -47,6 +43,24 @@ interface Database {
      */
     fun getNearbyChallenge(location: Location): Task<List<Challenge>>
 
+    /**
+     * Returns the followers of the user with the given user id.
+     *
+     * @param uid The user ID.
+     * @return A map where keys that are mapped to true indicates that it is a follower.
+     */
+    fun getFollowersOf(uid: String): Task<Map<String, Boolean>>
+
+    /**
+     * Makes the first user with the given uid follow the second user.
+     */
+    suspend fun follow(follower: String, followee: String)
```

Comment:
I believe we should either uses `Tasks` or coroutine but we should be consistent everywhere. The issue with coroutines is that when activities get destroyed the corresponding listeners is not removed (at least I think). As such I would recommend using Task<Unit> instead.

Reply:
Plus i hate the syntax 🦆

Reply:
As an addition to the comment above, I would like to say that your functions follow() and unfollow() are marked as 'suspend', but it is not clear why. It would be helpful to add a comment explaining why these functions need to be marked as 'suspend'.

Reply:
@Nicoslz, `suspend` functions (aka coroutines) are the Kotlin way of dealing with asynchrony. They are easier to use than callback-based solutions because they can be handled similarly to synchronous code, i.e. you write in a `suspend` function exactly how you would write if every operation was blocking.

While I do agree @BoyeGuillaume that we should probably not mix `suspend` functions and `Task`, I think that there is a case here to be made for keeping `follow` and `unfollow` as coroutines, since they are *one-shot* operations, and the `Task<Void>` alternative is definitely ugly, if not more difficult to code. Since there is roughly a one-to-one correspondence between `Task`s and coroutines, I would advocate for keeping them as coroutines, and plan in the future to use only coroutines—for their ease of use more than anything else.

Reply:
I checked the documentation and it seems you were right. Well my bad I guess, I don't have any other argument in favor of my beloved Task. I still believe that for function that fetches something we should use tasks instead (because of the codebase already present that ease the process of handling errors and cancellation)

Reply:
There will be further discussion regarding our choice of `Task` vs `Flow` when we will refactor the code in further sprints, for now I am comfortable with keeping everything as-is, and not try to migrate to either solution.